set.mm

```diff
@@ -741438,6 +741624,644 @@ corresponds to the definition of an (infinite) product in [Lang] p. 5,
+      VEVF $.
+  $}
+
+  $( Even the monoid of endofunctions on the empty set is a actually a monoid.
+     (Contributed by AV, 31-Jan-2024.) $)
+  efmnd0nmnd $p |- ( EndoFMnd ` (/) ) e. Mnd $=
+    ( c0 cvv wcel cefmnd cfv cmnd 0ex eqid efmndmnd ax-mp ) ABCADEZFCGAKBKHIJ
+    $.
+
+  $( The base set of the monoid of endofunctions on the empty set is the
+     singleton containing the empty set.  (Contributed by AV, 27-Jan-2024.) $)
+  efmndbas0 $p |- ( Base ` ( EndoFMnd ` (/) ) ) = { (/) } $=
+    ( vf c0 cmap co cv wceq cab cefmnd cfv cbs csn wf 0ex f0bi abbii eqtri eqid
+    mapval efmndbas df-sn 3eqtr4i ) BBCDZAEZBFZAGZBHIZJIZBKUBBBUCLZAGUEBBAMMRUH
+    UDAUCBNOPBUGUFUFQUGQSABTUA $.
+
+  ${
+    efmnd1bas.1 $e |- G = ( EndoFMnd ` A ) $.
+    efmnd1bas.2 $e |- B = ( Base ` G ) $.
+    ${
+      efmnd1bas.0 $e |- A = { I } $.
+      $( The monoid of endofunctions on a singleton has cardinality ` 1 ` .
+         (Contributed by AV, 27-Jan-2024.) $)
+      efmnd1hash $p |- ( I e. V -> ( # ` B ) = 1 ) $=
+        ( wcel chash cfv cexp co c1 cfn wceq csn snfi ax-mp syl5eq efmndhash cz
+        eqeltri fveq2i hashsng oveq12d 1z 1exp syl6eq ) DEIZBJKZAJKZULLMZNAOIUK
+        UMPADQZOHDRUCABCFGUASUJUMNNLMZNUJULNULNLUJULUNJKNAUNJHUDDEUETZUPUFNUBIU
+        ONPUGNUHSUIT $.
+
+      $d A p $.  $d I p $.  $d V p $.
+      $( The monoid of endofunctions on a singleton is consisting of the
+         identity only.  (Contributed by AV, 31-Jan-2024.) $)
+      efmnd1bas $p |- ( I e. V -> B = { { <. I , I >. } } ) $=
+        ( vp wcel csn cmap co cop cefmnd cfv fveq2i eqtri efmndbas cv wceq fsng
+        wf wb anidms snex elmap velsn 3bitr4g eqrdv syl5eq ) DEJZBDKZUMLMZDDNKZ
+        KZUMBCCAOPUMOPFAUMOHQRGSULIUNUPULUMUMITZUCZUQUOUAZUQUNJUQUPJULURUSUDDDE
+        EUQUBUEUMUMUQDUFZUTUGIUOUHUIUJUK $.
+    $}
+
+    ${
+      efmnd2bas.0 $e |- A = { I , J } $.
+      $( The monoid of endofunctions on a (proper) pair has cardinality ` 4 ` .
+         (Contributed by AV, 18-Feb-2024.) $)
+      efmnd2hash $p |- ( ( I e. V /\ J e. W /\ I =/= J ) -> ( # ` B ) = 4 ) $=
+        ( wcel w3a chash cfv cexp co c4 cfn wceq c2 wne eqeltri efmndhash ax-mp
+        cpr prfi fveq2i cvv elex id 3anim123i hashprb syl5eq oveq12d sq2 syl6eq
+        sylib ) DFKZEGKZDEUAZLZBMNZAMNZVCOPZQARKVBVDSADEUEZRJDEUFUBABCHIUCUDVAV
+        DTTOPQVAVCTVCTOVAVCVEMNZTAVEMJUGVADUHKZEUHKZUTLVFTSURVGUSVHUTUTDFUIEGUI
+        UTUJUKDEULUQUMZVIUNUOUPUM $.
+$(
+      @( The monoid of endofunctions on a pair is consisting of the identity,
+         the transposition and two mappings to the same element.  (Contributed
+         by AV, 18-Feb-2024.) @)
+      efmnd2bas @p |- ( ( I e. V /\ J e. W )
+                       -> B = ( { { <. I , I >. , <. J , J >. } ,
+                                { <. I , J >. , <. J , I >. } } u.
+                                { { <. I , I >. , <. J , I >. } ,
+                                { <. I , J >. , <. J , J >. } } ) ) @=
+        ? @.
+$)
```

Comment:
This proof is probably quite laborious. It is a good simple example but I'm not sure it is necessary.

Reply:
Yes, we would need a theorem similar to hash2prd:

```
    hash4qdd $p |- ( ( P e. V /\ ( # ` P ) = 4 )
                   -> ( ( ( ( W e. P /\ X e. P ) /\ ( Y e. P /\ Z e. P ) ) 
                            /\ ( ( W =/= X /\ W =/= Y /\ W =/= Z ) 
                              /\ ( X =/= Y /\ X =/= Z ) /\ Y =/= Z ) )
                          -> P = { W ,  X , Y , Z } ) ) $=
```